controllers/spacebindingrequestmigration/spacebindingrequest_migration_controller_test.go

```diff
@@ -0,0 +1,332 @@
+				Exists()
+			// the spaceBindingRequest wasn't created
+			spacebindingrequesttest.AssertThatSpaceBindingRequest(t, "jane-tenant", johnMur.Name+"-admin", member1.Client).
+				DoesNotExist()
+		})
+
+		t.Run("mur not found", func(t *testing.T) {
+			// given
+			member1 := NewMemberClusterWithClient(test.NewFakeClient(t), "member-1", corev1.ConditionTrue)
+			// let's not load the mur object
+			hostClient := test.NewFakeClient(t, janeMur, janeSpace, sbForCreator, sbForJohn)
+			ctrl := newReconciler(t, hostClient, member1)
+
+			// when
+			_, err = ctrl.Reconcile(context.TODO(), requestFor(sbForJohn))
+
+			// then
+			require.NoError(t, err)
+			// the spacebinding for the space creator is still there
+			spacebindingtest.AssertThatSpaceBinding(t, test.HostOperatorNs, janeMur.Name, janeSpace.Name, hostClient).
+				Exists()
+			// the spacebinding for john user is still there
+			spacebindingtest.AssertThatSpaceBinding(t, test.HostOperatorNs, johnMur.Name, janeSpace.Name, hostClient).
+				Exists()
+			// the spaceBindingRequest wasn't created
+			spacebindingrequesttest.AssertThatSpaceBindingRequest(t, "jane-tenant", johnMur.Name+"-admin", member1.Client).
+				DoesNotExist()
+		})
+
+		t.Run("spacebinding has spacebindingrequest", func(t *testing.T) {
+			// given
+			// the spacebinding has a spacebindingrequest
+			sbrForJohn := spacebindingrequesttest.NewSpaceBindingRequest("john-admin", "jane-tenant", spacebindingrequesttest.WithLabel(toolchainv1alpha1.SpaceCreatorLabelKey, "somevalue"))
+			sbForJohnWithSBR := spacebindingtest.NewSpaceBinding(johnMur.Name, janeSpace.Name, "admin", janeMur.GetName(), spacebindingtest.WithSpaceBindingRequest(sbrForJohn))
+			// let's not load the mur object
+			member1 := NewMemberClusterWithClient(test.NewFakeClient(t, sbrForJohn), "member-1", corev1.ConditionTrue)
+			hostClient := test.NewFakeClient(t, janeMur, janeSpace, sbForJohnWithSBR)
+			ctrl := newReconciler(t, hostClient, member1)
+
+			// when
+			_, err = ctrl.Reconcile(context.TODO(), requestFor(sbForJohnWithSBR))
+
+			// then
+			require.NoError(t, err)
+			// the spacebinding for john user is still there
+			spacebindingtest.AssertThatSpaceBinding(t, test.HostOperatorNs, johnMur.Name, janeSpace.Name, hostClient).
+				Exists()
+			// the spaceBindingRequest is unchanged
+			// no migration label as creator
+			spacebindingrequesttest.AssertThatSpaceBindingRequest(t, "jane-tenant", johnMur.Name+"-admin", member1.Client).
+				HasLabelWithValue(toolchainv1alpha1.SpaceCreatorLabelKey, "somevalue").
+				Exists()
+		})
+
+		t.Run("spacebinding is being deleted", func(t *testing.T) {
+			// given
+			member1 := NewMemberClusterWithClient(test.NewFakeClient(t), "member-1", corev1.ConditionTrue)
+			// the spacebinding is being deleted
+			hostClient := test.NewFakeClient(t, janeMur, janeSpace, sbForCreator, sbForJohn)
+			ctrl := newReconciler(t, hostClient, member1)
```

Comment:
you forgot to modify the SpaceBinding by adding the deletion timestamp
plus there is missing the john's MUR
```suggestion
			sbForJohn := spacebindingtest.NewSpaceBinding(johnMur.Name, janeSpace.Name, "admin", janeMur.GetName(), spacebindingtest.WithDeletionTimestamp())
			hostClient := test.NewFakeClient(t, janeMur, janeSpace, sbForCreator, johnMur, sbForJohn)
```

Reply:
🤦 good point! Thanks!